app/src/features/apiClient/contexts/apiClient.tsx

```diff
@@ -15,17 +15,16 @@ import {
 import { useTabsLayoutContext } from "layouts/TabsLayout";
 import { trackCreateEnvironmentClicked } from "../screens/environment/analytics";
 import PATHS from "config/constants/sub/paths";
-import { useLocation } from "react-router-dom";
 import useEnvironmentManager from "backend/environment/hooks/useEnvironmentManager";
-import { getEmptyAPIEntry } from "../screens/apiClient/utils";
+import { createBlankApiRecord } from "../screens/apiClient/utils";
 
 interface ApiClientContextInterface {
   apiClientRecords: RQAPI.Record[];
   isLoadingApiClientRecords: boolean;
   onNewRecord: (apiClientRecord: RQAPI.Record) => void;
   onRemoveRecord: (apiClientRecord: RQAPI.Record) => void;
   onUpdateRecord: (apiClientRecord: RQAPI.Record) => void;
-  onSaveRecord: (apiClientRecord: RQAPI.Record) => void;
+  onSaveRecord: (apiClientRecord: RQAPI.Record, redirectToTab?: boolean) => void;
```

Comment:
Why we need this `redirectToTab` ?

Reply:
If its a new record then it will be saved so we can open  the tab else, its an update so replace the tab